### PR TITLE
Completed ComplexityPractice Pair Assignment 

### PR DESCRIPTION
**Pull request by LauraV-702**
Laura V. and Diana K.

### CODE REVIEW

File: src/Practice.java [hunk 1]
```diff
@@ -70,15 +87,36 @@ public static int mostCommonTimeEfficient(int[] nums) {
...
+          count--;
+        }
+      
+      // Move to next number 
+        if (count == 0) {
+          mostCommon = nums[i];
+          count = 1;
+        }
+    }
+    return mostCommon;
```
Comment (auberonedu) [inline src/Practice.java:120]: This algorithm only works if we assume that there's one element that appears more than half the time. However, the code will fail for a case like this: `{3, 5, 7, 7, 2, 6, 8}`. This code will incorrectly claim that 8 is the most common element. Can you think of a new algorithm that properly handles this case?

File: src/PracticeTest.java [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+    int[] numArr1 = {1, 2, 3, 2, 4, 2};
+    System.out.println(Practice.mostCommonTimeEfficient(numArr1)); // 2
+  }
 
+  @Test
+  void testMostCommonSpaceEfficient() {
+    int[] numArr2 = {5, 5, 5, 5, 5, 5};
+    System.out.println(Practice.mostCommonSpaceEfficient(numArr2));
+  }
+
```
Comment (auberonedu) [inline src/PracticeTest.java:21]: These are good test cases, but they're not sufficient to thoroughly test the function. We'll often need to have several test cases to test each function.

File: src/PracticeTest.java [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
+  @Test
+  void testMostCommonTimeEfficient() {
+    int[] numArr1 = {1, 2, 3, 2, 4, 2};
+    System.out.println(Practice.mostCommonTimeEfficient(numArr1)); // 2
+  }
 
+  @Test
+  void testMostCommonSpaceEfficient() {
+    int[] numArr2 = {5, 5, 5, 5, 5, 5};
+    System.out.println(Practice.mostCommonSpaceEfficient(numArr2));
```
Comment (auberonedu) [inline src/PracticeTest.java:19]: This is a good edge case of checking how your method handles it if all the numbers are the same, but we should also test a more standard case like `{3, 5, 7, 7, 2, 6, 8};`

**Review by auberonedu — COMMENTED** [review]
Nice job with the complexity analysis! However, the algorithm for your implementation of the final method is flawed, and your tests aren't sufficient to expose this problem. Make sure to write more thorough tests in the future to catch these type of problems 🙂
